Fix part of #2862: create logical heading order

anthkris opened the pull request:
The following issues are addressed in this PR:

- No H1 on 404 page. First heading is an h4. Then it goes to h3. (Solution: The page now goes from h1 to h2)
- Category headings in the Library (and elsewhere) are h2s and then exploration "card" headings are also h2s, which doesn't provide a great hierarchy. (Solution: Card headings are now h3s)
- No H1 on Exploration page. (Solution: Exploration title in the navbar is now an h1)
- There is no h2 on this page (the about page, not the foundation or credits pages). Skips from h1 to h3. (Solution: Added an h1 with the text "About Oppia"; moved pronunciation pull-right element beneath it)
- At the beginning of the Contact Us page, we skip from h1 to h3. (Solution: Headings before the "Ways you can Help" heading are now h2s)
- There is no h1 on Donate page. The first heading is an h6, then goes to h2. (Solution: I reordered the divs so that the green sidebar with the "Donate to the Oppia Project" text comes first and changed those headings to h1 and h2. I tested on various window sizes and the page design remains in tact.)
- There is no h1 on Terms page. The first heading is an h2. (Solution: Changed headings to begin with h1; I did not preserve the styles as they seemed to just be defaults instead of specially chosen)
- There is no h1 on Privacy Policy page. The first heading is an h2. (Solution: Changed headings to begin with h1; I did not preserve the styles as they seemed to just be defaults instead of specially chosen)
- No h1 on Profile page. Skips to h2. (Solution: I updated the profile name in the navbar to be an h1, but I'm wondering if it would be better to update the profile name under the image to be the h1 instead. Thoughts?)

## Other Notes
- Unless otherwise noted, I updated the styles with the new headings to match the styles with the old headings.
- I noticed that the Collection page (the page you go to when you click on a collection in the library) also did not have an h1. I changed the current h2s to h1s. I did not update the styles. The current design looks different from the collection page on the production site and I wasn't sure if the design was finished. Thoughts?
- On the Terms of Use page, I noticed that several of the headings under the "What Information Does Oppia Collect?" section weren't headings at all; they were just bolded text. I changed them to headings of the appropriate order.

seanlip commented:
Thanks, @anthkris! For parts of this that you'd like UI comments on, please could you post screenshots to this thread?

anthkris commented:
@seanlip Sure thing:

For the profile page, I've made the page name in the navbar into an h1 (it looks the same as it always did):
<img width="1099" alt="screen shot 2017-01-23 at 5 06 17 pm" src="https://cloud.githubusercontent.com/assets/10858542/22226983/67cc28b4-e18e-11e6-8fc3-ee7eded2a25f.png">

Would it make more sense to just have the h1 in the profile name under the image?

On the collection page, I made the heading into an h1 from an h2:
<img width="1371" alt="screen shot 2017-01-23 at 5 08 09 pm" src="https://cloud.githubusercontent.com/assets/10858542/22227018/973ad82a-e18e-11e6-9e6d-b801f8fddc0a.png">

The heading is slightly bigger, but otherwise, the page looks the same. 

But the style is radically different from the current production site, even in terms of the alignment of the heading: 
<img width="1131" alt="screen shot 2017-01-23 at 5 09 28 pm" src="https://cloud.githubusercontent.com/assets/10858542/22227046/c646e2b2-e18e-11e6-93b4-97f80c2af8be.png">

So I wasn't sure if this was changing, and if so, if it was the finished product.

anthkris commented:
I should also note that the version of the collection page in dev is flagged by WAVE as having redundant links right next to each other (referring to the links in the table)

seanlip commented:
For the profile page, I think what you've done seems fine to me.

The collection page does look a little unaligned compared to the one in production. @kevintab95 is working on an update of that page, but in the meantime would you mind removing the links and capitalizing the objective in the objective column? Separate PR is fine too, and I think the change in title size is also fine (we may want to take another look at it again when the final version is ready).

seanlip commented:
@anthkris -- this looks fantastic, and I really appreciate the descriptions you've given of the various issues and how you've addressed them in the PR text. Thanks very much! LGTM, and ready to merge whenever you like.

anthkris commented:
@seanlip PTAL
I removed the link from the Objective side of the table on the collections page. I also added some javascript to capitalize the first letter of the first word of the objective. This seems to be dependent on whether or not the exploration author capitalized this word. I wasn't sure if the collections_player.html was the appropriate place to perform this operation.

seanlip commented:
Technically, not really; complicated JS doesn't belong in HTML. A cleaner way to do it would be to have a function like getCapitalizedObjective() defined on the CollectionNodeObjectFactory.js found in core/templates/dev/head/domain/collection.

anthkris commented:
Alrighty! Updated per your suggestion.

seanlip commented on core/templates/dev/head/domain/collection/CollectionNodeObjectFactory.js:
```diff
@@ -103,6 +103,12 @@ oppia.factory('CollectionNodeObjectFactory', [
     };
 
+    CollectionNode.prototype.getCapitalizedObjective = function(
+        explorationObjective) {
```
One more design issue -- this shouldn't take in an objective as an argument; instead, it should make use of the objective in this._explorationSummaryObject. Otherwise you potentially have two different sources of truth. (Think about it like this: someone external is asking this collection node for their 'capitalized objective', and this is something that the collection node should be able to derive on their own from the objective that they already 'know'.)

anthkris replied:
Okay. Should I use angular.copy or can I just use this._explorationSummaryObject.objective

anthkris replied:
Done

seanlip commented:
I don't think you need angular.copy here. Strings get copied anyway in js;
it's mainly arrays and objects which you need to be careful about .

On Wednesday, January 25, 2017, Kristin Anthony <notifications@github.com>
wrote:

> *@anthkris* commented on this pull request.
> ------------------------------
>
> In core/templates/dev/head/domain/collection/
> CollectionNodeObjectFactory.js <https://github.com/oppia/oppia/pull/2959>:
>
> > @@ -103,6 +103,12 @@ oppia.factory('CollectionNodeObjectFactory', [
>          explorationSummaryBackendObject);
>      };
>
> +    CollectionNode.prototype.getCapitalizedObjective = function(
> +        explorationObjective) {
>
> Okay. Should I use angular.copy or can I just use this._
> explorationSummaryObject.objective
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia/pull/2959>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKFeytw8T_l3zeERHQUP9pHoui43180zks5rV1xdgaJpZM4LrqFn>
> .
>

anthkris commented:
Very good. Done.

seanlip commented:
Thanks! LGTM (again :) )

anthkris commented:
Cool! I'll merge when all the checks pass.
